features/networking/service.feature

```diff
@@ -153,3 +153,325 @@ Feature: Service related networking scenarios
+      | ["spec"]["externalIPs"][0]  | <%= cb.hostip %> |
+    Then the step should succeed
+    """ 
+    Then I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get client command with:
+      | resource      | service               |
+      | resource_name | service-unsecure      |
+      | template      | {{.spec.externalIPs}} |
+      | o             | yaml                  |
+    And the output should contain "<%= cb.hostip %>"
+    """
+
+    # Create a pod
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/routing/caddy-docker.json |
+    Then the step should succeed
+    And the pod named "caddy-docker" becomes ready
+ 
+    # Curl externalIP:portnumber should pass
+    When I execute on the pod:
+      | /usr/bin/curl | -k | <%= cb.hostip %>:27017 |
+    Then the output should contain:
+      | Hello-OpenShift-1 http-8080                 |
+
+  # @author weliang@redhat.com
+  # @case_id OCP-24670
+  @admin
+  @destructive
+  Scenario: externalIP defined in service with set ExternalIP in rejectedCIDRs
+    Given 20 seconds have passed
+    Given I store all worker nodes to the :nodes clipboard
+    And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard
+
+    # Create additional network through CNO
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"rejectedCIDRs":["<%= cb.hostip %>/24"]}}}} |
+    Then I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | networks.config.openshift.io                  |
+      | resource_name | cluster                                       |
+      | template      | {{{{.spec.externalIP.policy.rejectedCIDRs}}}} |
+      | o             | yaml                                          |
+    And the output should contain "<%= cb.hostip %>"
+    """
+    #Clean-up required to erase above externalIP policy after testing done
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"rejectedCIDRs": null}}}} |
+    """
+
+    # Create a svc with externalIP
+    Given I have a project
+    And I wait up to 400 seconds for the steps to pass:
+    """
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0]  | <%= cb.hostip %> |
+    And the output should contain "externalIPs have been disabled"
+    """
+ 
+  # @author weliang@redhat.com
+  # @case_id OCP-24691
+  @admin
+  @destructive
+  Scenario: Defined Multiple allowedCIDRs 
+    Given 20 seconds have passed
```

Comment:
@weliang1 is this step needed at first?

Reply:
@zhaozhanqi This is new step added in this commit. I saw a new project will be created when "Given I store all worker nodes to the :nodes clipboard" and that project will be deleted after this test case finish. If next test case still has "Given I store all worker nodes to the :nodes clipboard" at the beginning and try to create a new project whose name will be the same as the one created in previous test case, then creating this new project will fail and whole secondary testing also fail, because that same name project still in terminated status. Using "Given 20 seconds" will give some time to clean previous project completely. Any two test cases running together and using "Given I store all worker nodes to the :nodes clipboard" will see this issue.

Reply:
@weliang1 having the tag `@destructive` means only run them sequentially.  So this step is not needed.

Reply:
@pruan-rht externalIP is policy configuration, any services' IP addresses which are not in policy allow CIDR will be blocked, so we should not execute any other test cases at same time.

Reply:
yes, if the case is `destructive`, then it will be run sequentially. 
@weliang1  you mean `"Given I store all worker nodes to the :nodes clipboard"` will create one project with specified project name. if so, how about check the project is not exist at the beginning.

Reply:
@zhaozhanqi 
First, we need "destructive" to let script run sequentially. 
Second, I do not really get what you asked.
"Given I store all worker nodes to the :nodes clipboard" is used for "And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard". This :hostip is needed and will not change in all new project created in following steps.

Reply:
I'm asking why we need this step "Given 20 seconds have passed". according to your last comment "I saw a new project will be created when "Given I store all worker nodes to the :nodes clipboard" and that project will be deleted after this test case finish. If next test case still has "Given I store all worker nodes to the :nodes clipboard" at the beginning and try to create a new project whose name will be the same as the one created in previous test case, then creating this new project will fail and whole secondary testing also fail, because that same name project still in terminated status"
So as your above reply, if so. we need to step "check the project is not exist at the beginning"

Reply:
@zhaozhanqi Move the test cases in different order, then we do not need "Given 20 seconds have passed" any more.

Reply:
ok. thanks.